rewindable-index/src/RewindableIndex/Storable.hs

```diff
@@ -231,10 +250,42 @@ getEvents
   -> StorableMonad h [StorableEvent h]
 getEvents s = do
   memoryEs <- getMemoryEvents (s ^. storage)
-              & V.freeze <&> V.toList
+              & V.freeze <&> foldEvents . V.toList
   diskEs   <- getStoredEvents (s ^. handle)
   pure $ diskEs ++ memoryEs
 
+{- This function is used to add a checkpoint to the in-memory part of event indexers.
+   You can use checkpoints to add synthetic events to the memory buffer. Filling the
+   memory buffer will flush the real events to disk faster than not using checkpoints.
+
+   This is useful if the number of events is small and you run the risk of never
+   storing them on-disk. When that happens, if the indexer is restarted it will
+   resume from the Genesis block, which is something that is not an expected
+   behaviour.
+
+   Possible future uses of checkpoints:
```

Comment:
Would it make sense to add a example using it in the `Spec/Sqlite` test module?

Reply:
I think the usage is fairly straight-forward. And I intend to base a lot more of the future improvements on this change. I would wait a bit until we write a new indexer specification so we get some more features in and then think about how we test them, what other properties we should add, etc.